### PR TITLE
add new pin shadow blocks and field editor

### PR DESCRIPTION
**Pull request by riknoll**
Fixes https://github.com/microsoft/pxt-microbit/issues/5733

This PR does a few things:

1. Creates three new abbreviated shadow blocks that only say `P0` instead of `analog pin P0`
2. Changes the shadow block for the `analog read` block so that it only lists the analog-readable pins
3. Adds a new field editor which displays a warning when a non-analog-readable pin is placed inside of the `analog read` block (see GIF below). This field editor is otherwise identical to the gridpicker
4. Renames the `pins.analogPin` and `pins.digitalPin` functions so that they start with an underscore, which means they get filtered out of the TS intellisense
5. Removes the `(write only)` text from the analog pin selector since the new field editor gives warnings instead

Here's what the warning looks like:

writeonly-pin

@sae220 FYI

### CODE REVIEW

**Comment by riknoll** [comment]
oh also, this PR relies on https://github.com/microsoft/pxt/pull/10118 but is safe to merge without it!

File: libs/core/pins.ts [hunk 1]
```diff
@@ -8,26 +8,79 @@ namespace pins {
...
      * Returns the value of a C++ runtime constant
      */
-    //% help=pins/digital-pin blockGap=8 advanced=true shim=TD_ID
-    //% blockId="digital_pin" block="digital pin %pin"
-    //% pin.fieldEditor="gridpicker" pin.fieldOptions.columns=4
+    //% help=pins/digital-pin
+    //% shim=TD_ID
+    //% blockId=digital_pin
+    //% block="digital pin $pin"
+    //% pin.fieldEditor=pinpicker
```
Comment (srietkerk) [inline libs/core/pins.ts:15]: I see that the field editor for the gridpicker had quotes around it, are the quotes not needed?

**Review by srietkerk — APPROVED** [review]
One question for my learning, but otherwise looks good!

File: libs/core/pins.ts (hunk 1 above)
Comment (riknoll) [reply libs/core/pins.ts:15]: nope! they don't hurt anything, i just like no quotes better

**Review by kimprice — APPROVED** [review]

**Comment by abchatra** [comment]
@carlosperate @jaustin please take a look at /beta and ensure all issues of pins are addressed as expected.

**Pull request merged**